### ISSUE TITLE
Mac OS Travis errors

### ISSUE THREAD

**Issue by sjakobi** [commercialhaskell/stack#2693]
Most of our Mac OS Travis jobs seem to be failing similarly to [this one](https://travis-ci.org/commercialhaskell/stack/jobs/166220713#L4101).

The relevant Travis issue seems to be this: https://github.com/travis-ci/travis-ci/issues/6522

I was hoping it would go away by itself, but maybe we'll have to use [a workaround](https://github.com/travis-ci/travis-ci/issues/6522#issuecomment-251734732).

**Comment by mgsloan** [comment]
Good point!  I've added the osx build to the list of "allowed_failures".  This issue also tracks removing it from that list.

**Comment by aryairani** [comment]
I think that travis issue is closed and doesn't occur anymore.  I'm here with a different mac build problem involving travis caching and `~/.stack/setup-exe-cache` corruption though, so it may be too soon to take osx off of `allowed_failures`, unless you disable caching too.
